turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2246,6 +2248,9 @@ async fn resolve_relative_request(
     let mut new_path = path_pattern.clone();
 
     if !fragment.is_empty() {
+        // 'fragments' should not be part of the result, however it could be that the user is
```

Comment:
this is the wrong place for this logic

fragment and query should only be extracted for patterns derived from imports, not require statements.

Then we wouldn't need to add the alternation since an import attempting to match a `#` sign would have to write `%23`

So that also means that import specifiers need to be parsed and then URI decoded.

Reply:
on the other hand of course we need to support queries and fragments in require to support loaders